pkg/infra/serverlock/serverlock.go

```diff
@@ -212,3 +252,16 @@ func (sl *ServerLockService) isLockWithinInterval(lock *serverLock, maxInterval
 	}
 	return false
 }
+
+func (sl ServerLockService) executeFunc(ctx context.Context, fn func(ctx context.Context)) {
+	start := time.Now()
+	ctx, span := sl.tracer.Start(ctx, "ServerLockService.executeFunc")
+	defer span.End()
+
+	ctxLogger := sl.log.FromContext(ctx)
+	ctxLogger.Debug("Start execution")
+
+	fn(ctx)
+
+	ctxLogger.Debug("Execution finished", "duration", time.Since(start))
```

Comment:
Does it make sense to have additional logs to capture the execution and duration of each function execution?

Reply:
I think this level of logging makes sense, it seems very relevant for me to be able to tell that most of the 14ms are overhead for the lock in this instance, and in another it might help to know the opposite.

Nitpicking, I'd like to see the action name here too, though it's not as important with the traceID present.

```
DEBUG[09-21|11:29:58] Execution finished                       logger=infra.lockservice traceID=69ca95db99002a7d00a207c1aa4f34e7 duration=439.175µs
DEBUG[09-21|11:29:58] LockExecuteAndRelease finished           logger=infra.lockservice traceID=69ca95db99002a7d00a207c1aa4f34e7 actionName="secret migration task " duration=14.408182ms
```